examples/codegen/tree/README.md

```diff
@@ -0,0 +1,65 @@
+# Tree example
+
+This example application reads its configuration from the KDB using code-generated functions.
+It then prints a tree defined inside the KDB.
+
+## Setup
+
+### CMake
+
+```sh
+# execute in the current directory or replace $PWD accordingly
+DIR=$PWD
+
+mkdir "$DIR/cmake/build" && cd "$DIR/cmake/build"
+
+cmake ..
+cmake --build .
+
+cd "$DIR"
+
+sudo kdb mount -R noresolver codegen_tree_example.conf "spec/sw/example/tree/#0/current" specload "app=$DIR/cmake/build/application"
+sudo kdb spec-mount "/sw/example/tree/#0/current"
+```
+
+### Pkgconfig
+
+```sh
+# execute in the current directory or replace $PWD accordingly
+DIR=$PWD
+
+cd "$DIR/pkgconfig"
+
+make
+
+cd "$DIR"
+
+sudo kdb mount -R noresolver codegen_tree_example.conf "spec/sw/example/tree/#0/current" specload "app=$DIR/pkgconfig/application"
+sudo kdb spec-mount "/sw/example/tree/#0/current"
+```
+
+## Running
+
+To run the application, simply execute:
+
+```sh
+# execute in the current directory or replace $PWD accordingly
+DIR=$PWD
+
+# for CMake
+LOC="cmake/build"
+
+# for Pkgconfig
+# LOC="pkgconfig"
+
+"$DIR/$LOC/application"
```

Comment:
Please also add some `kdb` commands to generate an example config. The output of the command with an empty `kdb`:

```
(empty)
```

 is pretty boring in my opinion. I assume that most people that use this tutorial will just copy and paste the commands here. Hence they probably will see the same output.